spec/Section 6 -- Execution.md

```diff
@@ -124,15 +128,25 @@ An initial value may be provided when executing a query.
 
 ExecuteQuery(query, schema, variableValues, initialValue):
 
+  * Let {subsequentPayloads} be an empty list.
   * Let {queryType} be the root Query type in {schema}.
   * Assert: {queryType} is an Object type.
   * Let {selectionSet} be the top level Selection Set in {query}.
   * Let {data} be the result of running
-    {ExecuteSelectionSet(selectionSet, queryType, initialValue, variableValues)}
+    {ExecuteSelectionSet(selectionSet, queryType, initialValue, variableValues, subsequentPayloads)}
     *normally* (allowing parallelization).
   * Let {errors} be any *field errors* produced while executing the
     selection set.
-  * Return an unordered map containing {data} and {errors}.
+  * Yield an unordered map containing {data} and {errors}.
+  * For each {payload} in {subsequentPayloads}:
+    * If {payload} is a Deferred Fragment Record:
+      * Yield the value from calling {ResolveDeferredFragmentRecord(payload, variableValues, subsequentPayloads)}.
+      * If {payload} is not the final payload in {subsequentPayloads}
+        * Add an entry to {payload} named `hasNext` with the value {true}.
+      * If {payload} is the final payload in {subsequentPayloads}
+        * Add an entry to {payload} named `hasNext` with the value {false}.
+    * If {payload} is a Stream Record:
+      * Yield ResolveStreamRecord(payload, variableValues, subsequentPayloads).
```

Comment:
this needs similar logic for when to include `hasNext` as the defer case

Reply:
Updated this so it should be clear that the logic applies to both the `defer` and `stream` case.